stdlib/src/mmap.rs

```diff
@@ -0,0 +1,1053 @@
+        }
+
+        #[inline]
+        fn advance_pos(&self, step: isize) {
+            self.pos.store(self.inner_pos() + step);
+        }
+
+        #[inline]
+        fn check_writable(&self, vm: &VirtualMachine) -> PyResult<()> {
+            if matches!(self.access, AccessMode::Read) {
+                return Err(
+                    vm.new_type_error("mmap can't modify a readonly memory map.".to_owned())
+                );
+            }
+
+            Ok(())
+        }
+
+        fn check_valid(&self, vm: &VirtualMachine) -> PyResult<PyMutexGuard<Option<MmapObj>>> {
+            let m = self.mmap.lock();
+
+            if m.is_none() {
+                return Err(vm.new_value_error("mmap closed or invalid".to_owned()));
+            }
+
+            Ok(m)
+        }
+
+        /// TODO: impl resize
+        #[allow(dead_code)]
+        fn check_resizeable(&self, vm: &VirtualMachine) -> PyResult<()> {
+            if self.exports.load() > 0 {
+                return Err(vm.new_buffer_error(
+                    "mmap can't resize with extant buffers exported.".to_owned(),
+                ));
+            }
+
+            if self.access == AccessMode::Write || self.access == AccessMode::Default {
+                return Ok(());
+            }
+
+            Err(vm.new_type_error(
+                "mmap can't resize a readonly or copy-on-write memory map.".to_owned(),
+            ))
+        }
+
+        #[pyproperty]
+        fn closed(&self) -> bool {
+            self.closed.load()
+        }
+
+        #[pymethod(magic)]
+        fn repr(zelf: PyRef<Self>, _vm: &VirtualMachine) -> PyResult<String> {
+            let mmap = zelf.mmap.lock();
+
+            if mmap.is_none() {
+                return Ok("<mmap.mmap closed=True>".to_owned());
+            }
+
+            let access_str = match zelf.access {
+                AccessMode::Default => "ACCESS_DEFAULT",
+                AccessMode::Read => "ACCESS_READ",
+                AccessMode::Write => "ACCESS_WRITE",
+                AccessMode::Copy => "ACCESS_COPY",
+            };
+
+            let repr = format!(
+                "<mmap.mmap closed=False, access={}, length={}, pos={}, offset={}>",
+                access_str,
+                zelf.len(),
+                zelf.inner_pos(),
+                zelf.offset
+            );
+
+            Ok(repr)
+        }
+
+        #[pymethod]
+        fn close(&self, vm: &VirtualMachine) -> PyResult<()> {
+            if self.closed() {
+                return Ok(());
+            }
+
+            if self.exports.load() > 0 {
+                return Err(vm.new_buffer_error("cannot close exported pointers exist.".to_owned()));
+            }
+            self.closed.store(true);
+            *self.mmap.lock() = None;
+
+            Ok(())
+        }
+
+        fn get_find_range(&self, options: FindOptions) -> (usize, usize) {
+            let pos = self.inner_pos();
+            let size = self.inner_size();
```

Comment:
```suggestion
            let size = self.inner_size().try_into().unwrap();
```
We don't need to keep duplicated `size.try_into().unwrap()` for next lines

Reply:
We need size's type to be `isize` for `end` assigment. But i can remove this duplicated invocation.